src/segment_owd.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type owm struct {
+	props       *properties
+	env         environmentInfo
+	temperature float64
+	weather     string
+}
+
+const (
+	APIKEY   Property = "apikey"
+	LOCATION Property = "location"
+	UNITS    Property = "units"
+)
+
+type weather struct {
+	ShortDescription string `json:"main"`
+	Description      string `json:"description"`
+	TypeID           string `json:"icon"`
+}
+type temperature struct {
+	Value float64 `json:"temp"`
+}
+
+type owmDataResponse struct {
+	Data        []weather `json:"weather"`
+	temperature `json:"main"`
+}
+
+func (d *owm) enabled() bool {
+	err := d.setStatus()
+	return err == nil
+}
+
+func (d *owm) string() string {
+	return fmt.Sprintf("%s (%g°)", d.weather, d.temperature)
+}
+
+func (d *owm) setStatus() error {
+	apikey := d.props.getString(APIKEY, ".")
+	location := d.props.getString(LOCATION, "De Bilt,NL")
+	units := d.props.getString(UNITS, "standard")
+
+	url := fmt.Sprintf("http://api.openweathermap.org/data/2.5/weather?q=%s&units=%s&appid=%s", location, units, apikey)
+	body, err := d.env.doGet(url)
+	if err != nil {
+		return err
+	}
+	q := new(owmDataResponse)
+	err = json.Unmarshal(body, &q)
+	if err != nil {
+		return err
+	}
+	d.temperature = q.temperature.Value
+	icon := ""
+	switch q.Data[0].TypeID {
+	case "01d":
+		icon = "滛"
```

Comment:
Can you add the string based representation of the icons? We always use `\uxxxx` for this. This will also make it so they aren't adjustable, could run into issues based on the font used. For our `os` segment we exposed all platform icons for example.

Reply:
Updated in 538ab69